pkg/log/zap/zap.go

```diff
@@ -207,3 +208,39 @@ func NewRaw(opts ...Opts) *zap.Logger {
 	log = log.WithOptions(o.ZapOpts...)
 	return log
 }
+
+// BindToFlagSet func allows CLI passed flags to be parsed.
+func (o *Options) BindFlags(fs *pflag.FlagSet) {
+	// Set Development mode value
+	fs.BoolVar(&o.Development, "zap-devel", false, "Enable zap development mode (changes defaults to console encoder, debug log level, and disables sampling)")
+
+	// Set Encoder value
+	var encVal encoderValue
+	encVal.setFunc = func(fromFlag zapcore.Encoder) {
+		o.Encoder = fromFlag
+	}
+	fs.Var(&encVal, "zap-encoder", "Zap log encoding ('json' or 'console')")
+
+	// Set the Log Level
+	var levelVal levelValue
+	levelVal.setFunc = func(fromFlag zap.AtomicLevel) {
+		o.Level = &fromFlag
+	}
+	fs.Var(&levelVal, "zap-log-level", "Zap log level (one of 'debug', 'info', 'error' or any integer value > 0)")
+
+	// Set the StrackTrace Level
+	var stackVal stackTraceValue
+	stackVal.setFunc = func(fromFlag zap.AtomicLevel) {
+		o.StacktraceLevel = &fromFlag
+	}
+	fs.Var(&stackVal, "zap-stacktrace-level", "Zap log level (one of 'warn',  'error'")
+
+}
+
+// UseFlagOptions to set logger with CLI passed flags.
+func UseFlagOptions(in *Options) func(o *Options) {
```

Comment:
@bharathi-tenneti How is this being used?
An example in the godocs for this would help since this exported. Plus I think we have an alias defined for the functional option type `func(o *Options)`:
```suggestion
func UseFlagOptions(in *Options) Opts {
```
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/log/zap/zap.go#L73